scanner/scanner.go

```diff
@@ -8,21 +8,26 @@ import (
 	"github.com/brimsec/zq/zng"
 )
 
+// Scanner implements the proc.Proc interface
 type Scanner struct {
 	reader zbuf.Reader
 	filter filter.Filter
 	span   nano.Span
 }
 
-func NewScanner(reader zbuf.Reader, f filter.Filter) *Scanner {
+func NewScanner(reader zbuf.Reader) *Scanner {
 	return &Scanner{
 		reader: reader,
-		filter: f,
+		span:   nano.MaxSpan,
 	}
 }
 
-func (s *Scanner) SetSpan(span nano.Span) {
-	s.span = span
+func NewFilteredScanner(reader zbuf.Reader, f filter.Filter, s nano.Span) *Scanner {
```

Comment:
Nit: I think Scanner should have just have one constructor:
```go
func NewScanner(reader zbuf.Reader, f filter.Filter, s nano.Span) *Scanner {
```

Reply:
I had that at one point, then thought it wasn't nice to have all the call sites with nil arguments... but was pretty ambivalent abuot which was better, so I'll switch back to single constructor.

Reply:
👍